lib/src/upgrader.dart

```diff
@@ -277,19 +277,29 @@ class Upgrader {
         var count = appcast.items == null ? 0 : appcast.items!.length;
         print('upgrader: appcast item count: $count');
       }
+      final criticalUpdateItem = appcast.criticalUpdateItem();
       final bestItem = appcast.bestItem();
       if (bestItem != null &&
           bestItem.versionString != null &&
           bestItem.versionString!.isNotEmpty) {
         if (debugLogging) {
           print(
               'upgrader: appcast best item version: ${bestItem.versionString}');
+          print(
+              'upgrader: appcast critical update item version: ${criticalUpdateItem?.versionString}');
         }
-        _appStoreVersion ??= bestItem.versionString;
-        _appStoreListingURL ??= bestItem.fileURL;
-        if (bestItem.isCriticalUpdate) {
+
+        _installedVersion ??= (await PackageInfo.fromPlatform()).version;
+
+        final criticalVersion = criticalUpdateItem?.versionString ?? '';
```

Comment:
This could be set right after `criticalUpdateItem` is set to keep this logic together.

Reply:
Updated https://github.com/larryaasen/upgrader/pull/275/commits/9075a49b424345db9a1c741756e77f3e8b35e986